api/src/main/java/uk/gov/dvsa/mot/trade/read/core/TradeReadServiceDatabase.java

```diff
@@ -130,12 +130,13 @@ public TradeReadServiceDatabase(MotTestReadService motTestReadService, VehicleRe
     public uk.gov.dvsa.mot.trade.api.Vehicle getLatestMotTestByRegistration(String registration) {
 
         List<Vehicle> vehicles = vehicleReadService.findByRegistration(registration);
+        VehicleAndLatestMot vehicleAndLatestMot = getVehicleAndLatestMotTestPass(vehicles);
 
-        if (CollectionUtils.isNullOrEmpty(vehicles)) {
+        if (vehicleAndLatestMot == null || !vehicleAndLatestMot.hasMotTest()) {
             return getDvlaVehicleByRegistration(registration);
```

Comment:
we are quering twice for vehicle, here (line 136) and in line 133. is it possible to avoid it?

Reply:
Good spot. I think we can replace line 136 with:
`return getLatestDvlaVehicleAndMapToTradeVehicle(vehicles);`

Reply:
Not sure how we can avoid this: The first query is for a DVSA vehicle. The second is for a DVLA vehicle.